interfaces/devdoc/malloc_multi_flex_requirements.md

```diff
@@ -0,0 +1,135 @@
+# `MALLOC_MULTI_FLEX` requirements
+
+## Overview
+
+`MALLOC_MULTI_FLEX` is a set of macros that allow allocating memory for a structure containing multiple variable sized array members using a single malloc.
+
+## Exposed API
+
+```c
+#define DECLARE_MALLOC_MULTI_FLEX(type, fields, array_fields)
+    ...
+    
+#define DEFINE_MALLOC_MULTI_FLEX(type, fields, array_fields)
+    ...
+
+#define MALLOC_MULTI_FLEX(type)
+    ...
+```
+
+## Example usage
+
+Lets say the user wants to allocate memory for a structure with name: `PARENT_STRUCT` which has three array members and three non-array members:
+- array_1 of type `uint32_t`
+- array_2 of type `uint64_t`
+- array_3 of type `INNER_STRUCT`
+- int_1 of type uint64_t
+- int_2 of type uin32_t
+- int_4 of type uint32_t
+
+where inner struct is another struct with its own members
+
+```c
+typedef struct INNER_STURCT_TAG
+{
+    uint32_t inner_int_1;
+    uint64_t inner_int_2;
+}INNER_STRUCT;
+```
+
+The caller first needs to specify the struct name, member types and names using `DECLARE_MALLOC_MULTI_FLEX` macro to define the struct and declare the malloc function:
+
+```c
+DECLARE_MALLOC_MULTI_FLEX(PARENT_STRUCT,
```

Comment:
>DECLARE_MALLOC_MULTI_FLEX [](http://example.com/codeflow?start=0&length=25)

Looking at this name it does not really tell me we have to deal with a struct. Maybe (just maybe) it would be a good thing to contain in its name the word "struct" somehow. Not married to the idea or what the name should be, just a Shakespearian consider

Reply:
Fixed